### PR TITLE
feat: add typings for stage instance

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds typings for the Stage Instances.

**Reference Discord API Docs PRs or commits:**
- [#2898](https://github.com/discord/discord-api-docs/pull/2898)
- [#3120](https://github.com/discord/discord-api-docs/pull/3120)
- [#2933](https://github.com/discord/discord-api-docs/pull/2933)

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

File: payloads/v9/stageInstance.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+/**
+ * https://discord.com/developers/docs/resources/stage-instance#stage-instance-object-privacy-level
+ */
+export const enum StageInstancePrivacyLevel {
+	/**
+	 * The stage instance is visible publicly, such as on stage discovery
+	 */
+	PUBLIC = 1,
```
Comment (vladfrangu) [inline payloads/v9/stageInstance.ts:43]: We do enums in PascalCase

File: payloads/v9/stageInstance.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	/**
+	 * The stage instance is visible to only guild members
+	 */
+	GUILD_ONLY,
+}
+
+export interface APIInviteStageInstance {
+	topic: string;
```
Comment (vladfrangu) [inline payloads/v9/stageInstance.ts:51]: Are there no docs for this?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: payloads/v9/stageInstance.ts (hunk 2 above)
Comment (iShibi) [reply payloads/v9/stageInstance.ts:51]: I must have forgot, I'll add them. Also, am I suppose to write the same stuff for `v8` too? I was unsure so I just added the changes only in `v9`.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by advaith1** [comment]
i think this is missing the gw events https://github.com/discord/discord-api-docs/pull/2933

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-    "files.eol": "\n",
-    "typescript.tsdk": "node_modules/typescript/lib"
+	"files.eol": "\n",
```
Comment (vladfrangu) [inline .vscode/settings.json:2]: ?

File: gateway/v8.ts [hunk 4]
```diff
@@ -222,6 +223,9 @@ export const enum GatewayDispatchEvents {
 	MessageReactionRemoveEmoji = 'MESSAGE_REACTION_REMOVE_EMOJI',
 	MessageUpdate = 'MESSAGE_UPDATE',
 	PresenceUpdate = 'PRESENCE_UPDATE',
+	StageInstanceCreate = 'StageInstanceCreate',
```
Comment (vladfrangu) [inline gateway/v8.ts:226]: the enum values are wrong

File: gateway/v9.ts [hunk 5]
```diff
@@ -225,6 +226,9 @@ export const enum GatewayDispatchEvents {
 	MessageReactionRemoveEmoji = 'MESSAGE_REACTION_REMOVE_EMOJI',
 	MessageUpdate = 'MESSAGE_UPDATE',
 	PresenceUpdate = 'PRESENCE_UPDATE',
+	StageInstanceCreate = 'StageInstanceCreate',
```
Comment (vladfrangu) [inline gateway/v9.ts:229]: Same

File: rest/v8/index.ts [hunk 6]
```diff
@@ -659,6 +660,18 @@ export const Routes = {
...
 	},
+
+	/**
+	 * Route for:
+	 * - GET `/stage-instances/{channel.id}`
+	 * - POST `/stage-instances`
+	 * - PATCH `/stage-instances/{channel.id}`
+	 * - DELETE `/stage-instances/{channel.id}`
+	 */
+	stageInstance(channelID?: Snowflake) {
```
Comment (vladfrangu) [inline rest/v8/index.ts:671]: This should be split into `stageInstances` and `stageInstance`, like `channelMessage(s)`

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 3 above)
Comment (iShibi) [reply .vscode/settings.json:2]: It's the linter that formats it like that during commit.

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**